src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp

```diff
@@ -1200,12 +1200,14 @@ CTranslatorExprToDXL::PdxlnDynamicBitmapTableScan(
 	CDXLNode *filter_dxlnode = PdxlnFilter(pdxlnCond);
 
 	CExpression *pexprRecheckCond = (*pexprScan)[0];
+	pexprRecheckCond->SetMotionInputs(pexprScan->GetMotionInputs());
```

Comment:
Also instead many times do so things I suggest do it only once in function CTranslatorExprToDXL::CreateDXLNode
```c
	const ULONG arity = pexpr->Arity();
	for (ULONG ul = 0; ul < arity; ul++)
	{
		CExpression *pexprChild = (*pexpr)[ul];
		pexprChild->SetMotionInputsFrom(pexpr);
	}
```